test/e2e/service_serving_ca_bundle_exists_test.go

```diff
@@ -0,0 +1,71 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+
+	"github.com/kube-reporting/metering-operator/test/reportingframework"
+)
+
+func testReportOperatorServiceCABundleExists(t *testing.T, rf *reportingframework.ReportingFramework) {
+	// Attempt to grab the reporting-operator Pod in the rf.Namespace.
+	// Because this Pod is spun up using a Deployment instead of a Statefulset
+	// we need to list any Pods matching the app=reporting-operator label as we
+	// don't know the name ahead of time.
+	podList, err := rf.KubeClient.CoreV1().Pods(rf.Namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: "app=reporting-operator",
+	})
+	require.NoError(t, err, "failed to list the reporting-operator Pod in the %s namespace", rf.Namespace)
+	require.Len(t, podList.Items, 1, "expected the list of app=reporting-operator Pods in the %s namespace to match a length of 1", rf.Namespace)
+
+	// We expect there's only going to be one reporting-operator
+	// in a single namespace, so hardcode the reporting-operator Pod
+	// object to be the first item in the list returned.
+	pod := podList.Items[0]
+	// Start building up a command that we'll execute against the
+	// reporting-operator container to ensure that the service-ca-bundle
+	// exists in the correct location.
+	execCmd := "ls /var/run/configmaps/service-ca-bundle"
+
+	req := rf.KubeClient.CoreV1().
+		RESTClient().
+		Post().
+		Resource("pods").
+		Name(pod.Name).
+		Namespace(rf.Namespace).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: "reporting-operator",
+			Command:   strings.Fields(execCmd),
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(rf.KubeConfig, "POST", req.URL())
+	require.NoError(t, err, "failed to execute the REST request")
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: &stdoutBuf,
+		Stderr: &stderrBuf,
+		Tty:    false,
+	})
+	assert.NoError(t, err, "failed to stream the exec results")
```

Comment:
We should probably generalize this as a helper function of sorts that takes something like a `*rest.Config`, `corev1.PodExecOptions` and returns a stdout/stderr buffer and error that was encountered.